### PR TITLE
Recursion, chipData, and wire connection data added, README updated, new test file.

### PR DESCRIPTION
**Pull request by bill090**
Added the recursion, added TODO and explanation stuff to README.md, added new test file. So I wrote a very bad explanation of how the save files work in the README, added a TODO list that is probably very inaccurate, implemented the recursion but still haven't bothered to made the linking of the chips, and also made a new test file that will be used for the new recursion thing. And then the rest of the things are bug fixes and small edits.

### CODE REVIEW

**Comment by bill090** [comment]
So now I've added the actual code to get the full data, so now the code is entirely complete except for the wire_layout. But I think that's the links between the chips, which I have done, so I think we're all good here.

File: main.py [hunk 1]
```diff
@@ -58,13 +58,14 @@ def create_chip_file(chip_obj, file_name):
 # Notes:        Second argument MUST BE LEFT TO DEFAULT!!!
 #               It is used internally, as a saved version of
 #               original chip argument
+#               Partial credit to: @bill090
 def create_new_chip(chip, return_chip = None):
     builtin_components = ["AND", "NOT", "SIGNAL IN", "SIGNAL OUT"]
     other_components = []
 
-    # if its the first iteration of recursion then return chip = chip
-    if return_chip == None:
```
Comment (sk8terkyd) [inline main.py:86]: Wait why did you remove this? this was making it so that if this function was called by the user and not by recursion then the return chip would equal the input chip so we could modify it more afterwards

Comment (bill090) [reply main.py:86]: The recursion happens regardless of the return_chip input.

Comment (bill090) [reply main.py:86]: But you can bring it back and do more with it.
